tokenserver/assignment/sqlnode/sql.py

```diff
@@ -315,9 +337,18 @@ def allocate_user(self, service, email, generation=0, client_state='',
         params = {
             'service': service, 'email': email, 'nodeid': nodeid,
             'generation': generation, 'keys_changed_at': keys_changed_at,
-            'client_state': client_state, 'timestamp': timestamp
+            'client_state': client_state, 'timestamp': timestamp,
+            'migration_state': ''
         }
         res = self._safe_execute(_CREATE_USER_RECORD, **params)
+        uid = res.lastrowid
+        # Update info if the user is selected to go to the spanner cluster:
+        if spanner_candidate and self.lucky_user(uid):
+            service = "spanner"
+            nodeid = params['nodeid'] = self._spanner_node_id
+            params['migration_state'] = "ORIGINAL"
+            node = self._spanner_entry
```

Comment:
This overrides any explicit `node` argument provided by the caller, which seems like it could be confusing in future.

Reply:
Hmm, good point. Part of this is that I'm still not 100% sure how the tokenserver arguments are actually provided, and how often a new, unallocated user might be assigned to an existing node. I'll add logic to only possibly provision those users to spanner if no node is set.